src/engine/enginebuffer.cpp

```diff
@@ -929,18 +930,13 @@ void EngineBuffer::process(CSAMPLE* pOutput, const int iBufferSize) {
         if (!bCurBufferPaused) {
             // The fileposition should be: (why is this thing a double anyway!?
             // Integer valued.
-            double filepos_play_rounded = round(m_filepos_play);
-            if (filepos_play_rounded != m_filepos_play) {
-                qWarning() << __FILE__ << __LINE__ << "ERROR: filepos_play is not round:" << m_filepos_play;
+            double playFrame = m_filepos_play / kSamplesPerFrame;
+            double filepos_play_rounded = round(playFrame) * kSamplesPerFrame;
```

Comment:
It is a preview of a solution when we move to a playposition in frames. Currently kSamplesPerFrame is const 2. that may change as well. 

It also fixes a minor rounding error. Example a track has 6 Samples and you wish to seek to 58 %
The track was samples at 0% 33% and 66%. 100% is the end position.     
The seek should be done to the 3. frame position since 66% is closest to 58 %.
The desired sample index is 4.

The old code: 
6 * 0,58 = 3,48 
round -> 3
force even = 2

The new code 
6 / 2 * 0,58 = 1,74 
round -> 2
2 * 2 = 4